eth/fetcher/tx_fetcher.go

```diff
@@ -171,16 +173,18 @@ type TxFetcher struct {
 
 	underpriced *lru.Cache[common.Hash, time.Time] // Transactions discarded as too cheap (don't re-fetch)
 
+	counter int // counter used to assign arrival order to tx announcements
```

Comment:
Oh actually it won't, it will just break ordering for a limited time and then work again
1 < 2 < 3 > -3 < -2 < -1

I would prefer a uint64 here anyway

Reply:
yup better safe than sorry, fixed.